runtime/runtime/src/lib.rs

```diff
@@ -1815,20 +1822,22 @@ impl Runtime {
                 break;
             }
 
-            delayed_receipt_count += 1;
-            let receipt = processing_state
+            let receipt = if let Some(receipt) = processing_state
                 .delayed_receipts
                 .pop(&mut processing_state.state_update, &processing_state.apply_state.config)?
-                .expect("queue is not empty");
-            let receipt = receipt.into_receipt();
+            {
+                receipt.into_receipt()
+            } else {
+                // Break loop if there are no more receipts to be processed.
+                break;
+            };
 
+            delayed_receipt_count += 1;
```

Comment:
I know it's the wrong place, but can you also keep count of the filtered receipts and log it or put it in metrics? Maybe you can do it with some simple math on the upper bound len?

Reply:
Added this to the TODOs as well. But I think yes, it should be possible to get the number of filtered receipts
`initial_upper_bound_len - delayed_receipt_count - final_upper_bound_len`